python-sdk/tutorials/automl-with-azureml/forecasting-pipelines/auto-ml-forecasting-pipelines.ipynb

```diff
@@ -0,0 +1,816 @@
+    "    arguments=[\n",
+    "        \"--model_name\",\n",
+    "        model_name_str,\n",
+    "        \"--ouput_dataset_name\",\n",
+    "        output_ds_name,\n",
+    "        \"--test_dataset_name\",\n",
+    "        test_dataset.name,\n",
+    "        \"--target_column_name\",\n",
+    "        target_column_name,\n",
+    "        \"--output_path\",\n",
+    "        output_data,\n",
+    "    ],\n",
+    "    compute_target=compute_target,\n",
+    "    allow_reuse=False,\n",
+    "    runconfig=run_config,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inference_pipeline = Pipeline(ws, [inference_step])\n",
+    "inference_run = experiment.submit(inference_pipeline)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inference_run.wait_for_completion(show_output=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Get the predicted data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core import Dataset\n",
+    "\n",
+    "inference_ds = Dataset.get_by_name(ws, output_ds_name)\n",
+    "inference_df = inference_ds.to_pandas_dataframe()\n",
+    "inference_df.tail(5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Schedule Pipeline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This section is about how to schedule a pipeline for periodically predictions. For more info about pipeline schedule and pipeline endpoint, please follow this [notebook](https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/machine-learning-pipelines/intro-to-pipelines/aml-pipelines-setup-schedule-for-a-published-pipeline.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inference_published_pipeline = inference_pipeline.publish(\n",
+    "    name=\"OJ Inference Test\", description=\"OJ Inference Test\"\n",
+    ")\n",
+    "print(\"Newly published pipeline id: {}\".format(inference_published_pipeline.id))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If `test_dataset` is going to refresh every 4 weeks before Friday 16:00 and we want to predict every 4 weeks (forecast_horizon), we can schedule our pipeline to run everyday at 16:00 to get daily inference results. You can refresh your test dataset (a newer version will be created) periodically when new data is available (i.e. target column in test dataset would have values in the beginning as context data, and followed by NaNs to be predicted). The inference pipeline will pick up context to further improve the forecast accuracy."
```

Comment:
Do you mean the test_dataset refresh every day (instead of every 4 weeks) hence we need do daily forecast to utilize the newly available data?

Reply:
changed to every 4 weeks